tests/console/sshd.pm

```diff
@@ -144,9 +149,18 @@ sub run {
     # Restore the /etc/ssh/sshd_config
     assert_script_run 'cp /etc/ssh/sshd_config{_before,}';
 
+    # Kill $ssh_testman to stop all SSH sessions
+    assert_script_run "killall -u $ssh_testman || true";
+    wait_still_screen 3;
+
     record_info("Restart sshd", "Restart sshd.service");
     systemctl("restart sshd");
 
+    # bsc#1175310 bsc#1181308 - Detect serious errors as they can be invisible because sshd may silently recover
+    if (script_run("journalctl -b | grep -A6 -B24 'segfault\\|fatal'") == 0) {
```

Comment:
This will catch any segfault from the current boot. `journalctl -b -u sshd.service` should be enough. IMO it's better to use post_*_hooks for these checks in general and filter it according to priority.

Reply:
I don't think that `-u` is a good idea but I can maybe mark the timestamp of the beginning of the test instead.

Reply:
why not ? It should show you logs from a specific unit.

Reply:
Yes, but there might be other associated units like `pam` or `systemd-authd` failing.

Reply:
> Yes, but there might be other associated units like `pam` or `systemd-authd` failing.

Not sure about `systemd-authd`, but `pam` modules used by sshd should be visible in the service's output. Nevertheless, segfaults can come from other test cases, and you are checking the whole log from current boot. Thus, eventually you can die due to unrelated issues.

Reply:
You're right Martin. I added the `-u sshd.service`, created a subroutine for it and prepended it to `post_*_hook`.